### ISSUE TITLE
redundant or unnecessary config

### ISSUE THREAD

**Issue by qoijjj** [secureblue/hardened-chromium#2]

**Comment by qoijjj** [comment]
disable: OptimizationHints

shouldn't be needed, since OptimizationHintsFetching is disabled already

**Comment by qoijjj** [comment]
disable: OptimizationHintsFetchingAnonymousDataConsent

shouldn't be needed since OptimizationHintsFetching is disabled

**Comment by qoijjj** [comment]
"DefaultGeolocationSetting": 2,

shouldn't be needed since the default is already to ask

**Comment by qoijjj** [comment]
```
"AccessibilityImageLabelsEnabled": false,
```

defaults to false: 

https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/accessibility/accessibility_labels_service.cc;l=148?q=AccessibilityImageLabelsEnabled%20lang:cc&ss=chromium%2Fchromium%2Fsrc

**Comment by qoijjj** [comment]
`"NetworkServiceSandboxEnabled": true`

defaults to true:
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/net/system_network_context_manager.cc;l=682?q=NetworkServiceSandboxEnabled%20lang:cc&ss=chromium%2Fchromium%2Fsrc

**Comment by qoijjj** [comment]
`"WebRtcEventLogCollectionAllowed": false,`

defaults to false:


https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/browser_ui_prefs.cc;l=108?q=WebRtcEventLogCollectionAllowed%20browser_ui_prefs&ss=chromium%2Fchromium%2Fsrc

**Comment by qoijjj** [comment]
`"ImportSavedPasswords": false,`

https://admx.help/?Category=Chrome&Policy=Google.Policies.Chrome::ImportSavedPasswords

**Comment by qoijjj** [comment]
"DefaultInsecureContentSetting": 2,

mixed content is blocked by default:

https://source.chromium.org/chromium/chromium/src/+/main:chrome/app/settings_strings.grdp;l=3462?q=IDS_SETTINGS_SITE_SETTINGS_INSECURE_CONTENT_BLOCK&ss=chromium%2Fchromium%2Fsrc

**Comment by qoijjj** [comment]
"RemoteDebuggingAllowed": false,


Only relevant for headless, which we don't build: https://source.chromium.org/chromium/chromium/src/+/main:headless/lib/browser/headless_browser_impl.cc;l=343?q=IsRemoteDebuggingAllowed%20lang:cc&ss=chromium%2Fchromium%2Fsrc

Also, only relevant if the user deliberately enables it:

https://admx.help/?Category=Chrome&Policy=Google.Policies.Chrome::RemoteDebuggingAllowed

**Comment by qoijjj** [comment]
"SafeSitesFilterBehavior": 0,


Defaults to 0: https://admx.help/?Category=Chrome&Policy=Google.Policies.Chrome::SafeSitesFilterBehavior

**Comment by qoijjj** [comment]
"SpellCheckServiceEnabled": false,


disabled by default:

https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/spellchecker/spellcheck_factory.cc;l=66?q=spellcheck::prefs::kSpellCheckUseSpellingService%20lang:cc&ss=chromium%2Fchromium%2Fsrc

**Comment by qoijjj** [comment]
"UrlKeyedAnonymizedDataCollectionEnabled": false,


Contrary to the policy description, this appears to be disabled by default:

![image](https://github.com/user-attachments/assets/9253ffc4-1698-445a-a53d-1fabcb9cd52b)

**Comment by qoijjj** [comment]
"MediaRecommendationsEnabled": false,


policy was removed

https://source.chromium.org/chromium/chromium/src/+/main:components/policy/test/data/pref_mapping/MediaRecommendationsEnabled.json;l=3;bpv=1;bpt=0

**Comment by qoijjj** [comment]
"CloudPrintSubmitEnabled": false,

deprecated

![image](https://github.com/user-attachments/assets/d15bb1bc-24e6-4946-91bf-1b08a563352f)

**Comment by qoijjj** [comment]
"SyncDisabled": true,


not required, unused by default, opt-in required

**Comment by qoijjj** [comment]
> Don't want to open a new issue since this is relevent here. I'm not sure if `NetworkServiceSandboxEnabled` true by default. I've attached some screenshots to show what I mean. This is the policy absent (process ID check using 

very strange. if this is the case then I'll need to dig through the code more

**Comment by qoijjj** [comment]
@RKNF404 The only reason I can see from the code why the network sandbox would be disabled is if you left the policy in place but with a non-true value:

https://chromium-review.googlesource.com/c/chromium/src/+/4702341

**Comment by qoijjj** [comment]
> If this policy is not set, the default configuration for the network sandbox will be used. This may vary depending on Google Chrome release, currently running field trials, and platform. 

@RKNF404 are you running secureblue? what policies and chromium.conf are you running?

**Comment by qoijjj** [comment]
> I definitely double check with code definitions, but it's hard to definitvely say sometimes. Safe than sorry, no harm done if there is room for doubt. But that's my mentality. There are cases where features are disabled (or enabled) and they do not coincide with the browsers behavior, especially code defined defaults.

Agreed, we should test thoroughly. 

> I'm on Workstation, running my own conf file. The feature to toggle the sandbox is not used either. I do publish my policies and flags if you want me to link it. But, I don't think what I do makes a difference.

The only reason I asked is cause the policy mentions field trials. 

I will test these more once the build completes.

**Comment by qoijjj** [comment]
accidentally edited instead of responding

> you can test it that way short term, since it is the basis.

yep i will

> I guess the ChromeVariations policy? Maybe?

yeah i'm not sure. just spitballing

**Comment by qoijjj** [comment]
@RKNF404 I think I found a potential cause of the discrepancy you're seeing. I just ran checksec against two different chromium processes for the same chromium instance with the network sandbox enabled, and one showed `   No Seccomp `, while the other showed ` Seccomp-bpf `. 

So it might be that the network change was not the cause of the difference in seccomp results, and the results are a red herring

**Comment by qoijjj** [comment]
if you run `checksec --proc-all` you'll see that some chromium processes show seccomp, others don't, regardless of network sandbox policy.

@RKNF404

**Comment by qoijjj** [comment]
actually wait, you said you filtered on Network Service, I missed that.

let me try that

**Comment by qoijjj** [comment]
```
5408   /usr/lib64/chromium-browser/chromium-browser --type=utility --utility-sub-type=network.mojom.NetworkService
```

```
# checksec --proc=5408


         COMMAND    PID RELRO           STACK CANARY            SECCOMP          NX/PaX        PIE                     FORTIFY
 chromium-browse   5408 Full RELRO      Canary found            No Seccomp       NX enabled    PIE enabled             Yes

```

![image](https://github.com/user-attachments/assets/62a987cc-f97c-4000-87e0-f648767f6ee5)


Nope, I'm seeing No Seccomp for the NetworkService process with the policy enabled @RKNF404

**Comment by qoijjj** [comment]
interesting. it looks like field trials of the network service sandbox only started this year for windows:

https://chromium-review.googlesource.com/c/chromium/src/+/5260025

**Comment by qoijjj** [comment]
hmm, but this config is already present:

```
   "NetworkServiceSandboxLinuxAndChromeOS": [
        {
            "platforms": [
                "linux",
                "chromeos",
                "chromeos_lacros"
            ],
            "experiments": [
                {
                    "name": "EnabledWithMitigation",
                    "enable_features": [
                        "NetworkServiceFileAllowlist",
                        "NetworkServiceSandbox",
                        "NetworkServiceSyscallFilter"
                    ],
                    "disable_features": [
                        "kForceDisableSpectreVariant2MitigationInNetworkService"
                    ]
                }
            ]
        }
    ],
    ```

**Comment by qoijjj** [comment]
@RKNF404 I think I found it. It's disabled by default contrary to the other code:

https://source.chromium.org/chromium/chromium/src/+/main:sandbox/policy/features.cc;l=23?q=NetworkServiceSandbox%20lang:cc

**Comment by qoijjj** [comment]
To double check, I tried with both suid and namespace sandboxing, both with the NetworkServiceSandboxEnabled policy set to true, both showing the following for the network.mojom.NetworkService

```


* Does the CPU support NX: Yes

         COMMAND    PID RELRO           STACK CANARY            SECCOMP          NX/PaX        PIE                     FORTIFY
 chromium-browse   5417 Full RELRO      Canary found            No Seccomp       NX enabled    PIE enabled             Yes
 ```

@RKNF404 can you double check your findings?

**Comment by qoijjj** [comment]
> Huh... strange. On the vanilla Fedora build? Did you try enabling the sandboxing features? Like `NetworkServiceSyscallFilter`? Without enabling `NetworkServiceSandbox`?\

This is the vanilla fedora build with secureblue policies and config


> NetworkServiceSyscallFilter

is this a policy?

it's enabled by default regardless: https://source.chromium.org/chromium/chromium/src/+/main:sandbox/policy/features.cc;l=31?q=NetworkServiceSandbox%20lang:cc

**Comment by qoijjj** [comment]
> @qoijjj It's a feature, it's just a test to see if it is being enabled but seccomp itself isn't, or the policy isn't enabling the sandbox at all.

I see, so you're passing it in --enable-features?

**Comment by qoijjj** [comment]
@RKNF404 how are you verifing the seccomp state is the same without checksec? or you used it but just didn't record it?

**Comment by qoijjj** [comment]
also @RKNF404 are you on discord? would be easier :smile:

**Comment by qoijjj** [comment]
@RKNF404 message me in #dev, I don't know what your username is

**Comment by qoijjj** [comment]
@RKNF404 confirmed in code, spellcheck service is disabled by default:

https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/spellchecker/spellcheck_factory.cc;l=69?q=kSpellCheckUseSpellingService%20lang:cc&ss=chromium%2Fchromium%2Fsrc&start=1

**Comment by qoijjj** [comment]
@RKNF404 same with image labels: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/accessibility/accessibility_labels_service.cc;l=148?q=kAccessibilityImageLabelsEnabled%20lang:cc&ss=chromium%2Fchromium%2Fsrc

**Comment by qoijjj** [comment]
me neither. with sync though I did more digging into the logic, and it's more nuanced. It looks like sync might be enabled by default according to this function:

https://source.chromium.org/chromium/chromium/src/+/main:components/sync/base/command_line_switches.cc;l=14;drc=4cc063ac39c4a0d1f6011421b259a9715bb16de1;bpv=1;bpt=1

I'm gonna invert the logic and make it return true only if "--enable-sync" is explicitly passed

**Comment by qoijjj** [comment]
@RKNF404 network service: https://github.com/secureblue/hardened-chromium/commit/3c6e77ab52b09afbcb1074ceb3f8e70efb334503

sync: https://github.com/secureblue/hardened-chromium/commit/50d6ab90f4fe2361748b9cfbf0e4eed597877dca
